dataset-processor-python/data_loader.py

```diff
@@ -0,0 +1,100 @@
+import os
+import sys
+import csv
+import json
+
+
+class DataLoader:
+    def __init__(self, path, dataset_name):
+        self.path = path
+        self.dataset_name = dataset_name
+
+    def csv_to_json(self):
+        """
+        converts the csv file to json format
+        """
+        try:
+            with open(self.path, "r") as f:
+                reader = csv.DictReader(f)
+                data = list(reader)
+            json_filename = self.dataset_name + ".json"
+            full_path = os.path.join(os.path.dirname(self.path), json_filename)
+            with open(full_path, "w") as f:
+                json.dump(data, f, indent=4)
+        except Exception as e:
+            print(f"Error while converting CSV to JSON: {e}")
+
+
+class DatasetManager(DataLoader):
+    """
+    Class to create the dataset folder and json files
+    """
+
+    def __init__(self, path, dataset_name):
+        super().__init__(path, dataset_name)
+
+    def get_inputs(self):
+        """
+        get the inputs from the user
+        """
+        # version is required, title and description are optional
+        # Q: should we have a default title and description?
+        # return version, title, description
+        pass
```

Comment:
Is this getting command line inputs (prompting for user text) or from parameters?

Reply:
As of now, we are gathering user inputs through the command line prompts. We might revise this approach depending on how much data is required from the user.